File: DesignPatternsExercises/java/IteratorDesignPattern/Solution/IteratorDesignPattern.java
import java.util.Iterator;

public class IteratorDesignPattern {
    public static void main(String[] args) {
        Iterator iterator;

        System.out.println("Array of Countries:");
        iterator = new CountryArray().getIterator();
        displayCountries(iterator);

        System.out.println("\nList of Countries:");
        iterator = new CountryList().getIterator();
        displayCountries(iterator);

        System.out.println("\nSet of Countries:");
        iterator = new CountrySet().getIterator();
        displayCountries(iterator);
        System.out.println();
    }

    // public static void __________(1)__________(Iterator iterator) {
    public static void displayCountries(Iterator iterator) {
        while (iterator.hasNext()) {
            // Country country = __________(2)__________ iterator.next();
            Country country = (Country) iterator.next();
            System.out.println("Country: " + country.getName() + " -- Area: " + country.getArea() +
                    " -- Population: " + country.getPopulation());
        }
    }
}

File: DesignPatternsExercises/ruby/iterator_design_pattern/exercise/iterator_design_pattern.rb
require_relative 'country_array'
require_relative 'country_set'

def __________(1)__________
  __________(2)__________.each { |country| puts "Country: #{country.name} -- Area: #{country.area} -- Population: #{country.population}" }
end

puts 'Array of Countries:'
@iterator = CountryArray.new.__________(3)__________
display_countries
puts ''

puts 'Set of Countries:'
@iterator = CountrySet.new.__________(4)__________
display_countries
